sound/soc/sof/sof-of-dev.c

```diff
@@ -83,7 +84,7 @@ static int sof_of_probe(struct platform_device *pdev)
 	if (mach)
 		mach->mach_params.platform = dev_name(dev);
 
-	sof_pdata->machine = mach;
+	sof_pdata->machine = (struct snd_soc_fw_mach *)mach;
```

Comment:
you have not changed the definition of sof_pdata, so why do you need to cast it to snd_soc_fw_mach here?

Reply:
Actually, I did change it :).